pylxd/client.py

```diff
@@ -570,7 +570,14 @@ def assert_has_api_extension(self, name):
     def authenticate(self, secret, use_token_auth=True):
         if self.trusted:
             return
-        cert = open(self.api.session.cert[0]).read().encode("utf-8")
+
+        if not isinstance(self.api.session.cert, tuple):
+            raise exceptions.ClientConnectionFailed("No client certificate specified")
+
+        try:
+            cert = open(self.api.session.cert[0]).read().encode("utf-8")
+        except FileNotFoundError:
+            raise exceptions.ClientConnectionFailed("Client certificate not found")
```

Comment:
If the purpose is just to check if the file exists, just `open(self.api.session.cert[0])` should suffice.
Also, I think we should close the file in case it exists.